utils/src/main/kotlin/ArchiveUtils.kt

```diff
@@ -69,34 +70,40 @@ enum class ArchiveType(vararg val extensions: String) {
 }
 
 /**
- * Unpack the [File] to [targetDirectory].
+ * Unpack the [File] to [targetDirectory] using [filter] to select only the entries of interest.
```

Comment:
commit: Why is logic necessary? The filtering can be applied as well after extracting ALL files for simplicity.
The commit message rationale should at least consider this IMO.

Reply:
Added a corresponding statement to the commit message, although IMHO the reasoning is pretty obvious.

Reply:
Well, it was obvious to me as well that the filtering is faster. I wasn't after this.I hope I don't just misunderstand this.
I wanted to understand the use case making it necessary. So you say you are planning to filter for LICENSE files.
and NOTICE files. Can you give a hint (not necessarily in the commit message) what the use case is?